pkg/router/route.go

```diff
@@ -0,0 +1,66 @@
+package router
```

Comment:
i kinda question if this needs to be its own package versus another file in config.  What drew you to make it a package?

Reply:
I got some cycle references; at the beginning I moved some code to the `cmd/content-sources/main.go` and it resolved, but I was not happy with that solution. Visiting some repository I saw that the routing for the api endpoints where set in a different package; at the beginning I thought "ey! I like how it looks", and I saw it resolved the dependency; but a deeper thinking reveals what solve the situation:

- in golang the dependency granularity happens at package level, so moving the code into a file in the same package would improve the clashing with changes into the repository, but it would not fix the dependency situation.
- the situation is that some code in `pkg/config` depends on `pkg/handler` and `pkg/handler` depends on `pkg/config`.
- The code that configure the route has the dependency on `pkg/handler` (and `pkg/middleware`), so when I extracted the code into a new package, now I have the below dependencies:
  - `pkg/routes` > `pkg/config`, `pkg/handler`, `pkg/middleware`, `pkg/instrumentation`, `pkg/client` (I moved them out of `pkg/config`).
  - `pkg/handler` > `pkg/config`, ...  (but not on `pkg/routes`).
  - `pkg/config` > `pkg/errors`, `pkg/event` (I can see a refactor here, moving back the kafka configuration to `pkg/config`, as I think that probably `pkg/event` now can depends on `pkg/config` with no cycle references; the tool that @rverdile mentioned recently will arise a better big picture, I have to try it).
- `pkg/config` is broadly referenced into the repository (below what I get with a quick `grep -HRn pkg/config pkg/`):

```raw
pkg/api
pkg/handler
pkg/models
pkg/external_repos
pkg/middleware
pkg/seeds
pkg/db
pkg/dao
pkg/route
```

So with moving the code to a new package, many risky cycle dependencies from `pkg/config` have been extracted.

Thanks for the question, that let me think deeper about this.

> Probably I am missing some dependency into the above, I didn't check all the import for the files at `pkg/config` package.

Reply:
I see the dependency on `pkg/handler` in `pkg/router`, but I'm curious why handler needs to be called there. `cmd/main.go` is registering the routes already, so why does it need they also need to be registered in the `ConfigureEcho()`?

With that said `pkg/router` would still be needed to avoid a circular dependency. There would be a circular dependency between `pkg/config` and `pkg/middleware` without `pkg/router`. 

I've been trying some ideas, but I'm not sure really sure what the best way to handle that is. If we want a config package, a middleware package, but no third package to handle the circular dependency, maybe the middleware can be registered in `cmd/main.go` after `ConfigureEcho() is called.